tools/autograd/gen_variable_type.py

```diff
@@ -80,6 +80,28 @@
     '_coalesced_',
 }
 
+# When a function modifies its input tensors, it should never change the the input tensors'
+# underlying c10::TensorImpl pointers or c10::Storage pointers.
+#
+# The following code templates
+# - `SAVE_TENSOR_STORAGE`
+# - `ENFORCE_SAME_TENSOR_STORAGE`
+# - `SAVE_TENSORLIST_STORAGE`
+# - `ENFORCE_SAME_TENSORLIST_STORAGE`
+# - `SAVE_TENSOR_IMPL`
+# - `ENFORCE_SAME_TENSOR_IMPL`
+# - `SAVE_TENSORLIST_IMPL`
+# - `ENFORCE_SAME_TENSORLIST_IMPL`
+# implement the checks for this invariant.
+#
+# The following list contains functions that we don't enforce the invariant on.
+DONT_ENFORCE_SAME_TENSOR_IMPL_OR_STORAGE = {
+    # These functions are expected to change impl or storage of input tensors
+    '_th_set_', '_cudnn_rnn_flatten_weight',
```

Comment:
are you going to change these?

Reply:
It's likely not possible to fix `_th_set_` because changing storage is exactly its purpose. For fixing `_cudnn_rnn_flatten_weight`, I opened an issue at https://github.com/pytorch/pytorch/issues/16695.